newrelic/newrelic-cli
# nrdiag integration

Integrates `nrdiag` into the `newrelic` CLI by downloading & running it with some specially-chosen parameters for different situations.

# To test

Build & run `bin/{platform}/newrelic diagnose run`. This will download the `nrdiag` distribution file at `https://download.newrelic.com/nrdiag/nrdiag_latest.zip`, extract the binary appropriate for your system, and place it in `.newrelic/bin` in your home directory, then run it with the default parameters.

Run `newrelic diagnose lint --config-file newrelic.yml` to try the linter on a Java agent config file.

Run `newrelic diagnose update` to check for a newer version of `nrdiag` and download it if available.

Note: I haven't tested this on Windows yet.
